processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -402,7 +424,14 @@ public static ComparableList convertToList(Object obj)
     if (obj instanceof ComparableList) {
       return (ComparableList) obj;
     }
-    throw new ISE("Unable to convert type %s to %s", obj.getClass().getName(), ComparableList.class.getName());
+    if (obj instanceof Object[]) {
+      List<T> delegateList = new ArrayList<T>();
+      for (Object eachObj : (Object[]) obj) {
+        delegateList.add(convertFunction.apply(eachObj));
```

Comment:
This code is not symmetrical with the `List<>` case above.  Is there a specific reason why objects need conversion when they are an array and not when they are a `List<>`?  It seems like they should generally be the same code path, so if `return new ComparableList((List) obj);` is correct for List, why can it not also be correct for the Object[] case?  Or, vice-versa, if the converted form is correct, then I think the `List<>` case above needs update.

Do we know which code path tickles the `List<>` case above?  Is there a test that covers it?

Reply:
Yes, there is a test case for it. 
`GroupByQueryTestRunner#testGroupByWithLongArrays()`
on further inspection, its returning doubles in the output. That looks incorrect to me. 

Which means we might have to transform the list as well. Similar argument can be made for the String Array stuff as well. 

WDYT ?

Reply:
to be consistent with the behavior of other calls to `DimensionHandlerUtils.convertObjectToX`, which is used by the grouper to ensure everything is expected and consistent type (both to handle schema differences between segments and also handle jackson changing types like doubles to floats, longs to ints, etc), these methods should be coercing the elements I think, so the List case is probably wrong.

Reply:
Acked!!